File: Programmers/완주하지 못한 선수.py
def solution(participant, completion):
    dict = {}
    answer = ''
    for name in participant:
        if name in dict:
            dict[name] = dict[name] + 1
        else:
            dict[name] = 1

    for name in completion:
        dict[name] = dict[name] - 1

    for key, val in dict.items():
        if dict[key] > 0:
            answer = key

    return answer

File: Programmers JS/완주하지 못한 선수.js
function solution(participant, completion) {
  var answer = "";
  let people = {};
  // for(let person of participant){
  //     {person in people ? people[person]++ : people[person] = 1}
  // }
  // for(let person of completion){
  //     people[person] --;
  // }
  // for(let person in people){
  //     if(people[person] > 0){
  //         answer = person
  //     }
  // }

  participant.forEach((person) =>
    person in people ? people[person]++ : (people[person] = 1)
  );
  completion.forEach((person) => people[person]--);
  for (let person in people) {
    if (people[person] > 0) {
      answer = person;
    }
  }
  return answer;
}
